src/modules/core/apollo/apolloClient.ts

```diff
@@ -0,0 +1,100 @@
+import { useMemo } from 'react';
+import {
+  ApolloClient, HttpLink, InMemoryCache, ApolloLink, NormalizedCacheObject, FieldPolicy,
+} from '@apollo/client';
+import merge from 'deepmerge';
+import isEqual from 'lodash/isEqual';
+
+// Should we use "__APOLLO_STATE__" instead of "apolloState"?
+// Is a better pattern? next.js with-apollo uses it.
+export const APOLLO_STATE_PROP_NAME = 'apolloState';
```

Comment:
Might be better to use `__APOLLO_STATE__` to avoid page props clashing indeed.

Next.js uses `__X__` for every built-in props that shouldn't clash with custom props